scvi/dataset/dataset10X.py

```diff
@@ -1,201 +1,263 @@
-                       "t_4k",
-                       "neuron_9k"
-                       ],
-                      '3.0.0':
-                      ['pbmc_1k_protein_v3',
-                       'pbmc_10k_protein_v3',
-                       'malt_10k_protein_v3',
-                       'pbmc_1k_v2',
-                       'pbmc_1k_v3',
-                       'pbmc_10k_v3',
-                       'hgmm_1k_v2',
-                       'hgmm_1k_v3',
-                       'hgmm_5k_v3',
-                       'hgmm_10k_v3',
-                       'neuron_1k_v2',
-                       'neuron_1k_v3',
-                       'neuron_10k_v3',
-                       'heart_1k_v2',
-                       'heart_1k_v3',
-                       'heart_10k_v3'
-                       ]}
-
-to_groups = dict([(dataset_name, group)
-                  for group, list_datasets in available_datasets.items()
-                  for dataset_name in list_datasets])
+from scvi.dataset.dataset import DownloadableDataset, _download
+
+logger = logging.getLogger(__name__)
+
+available_datasets = {
+    "1.1.0": [
+        "frozen_pbmc_donor_a",
+        "frozen_pbmc_donor_b",
+        "frozen_pbmc_donor_c",
+        "fresh_68k_pbmc_donor_a",
+        "cd14_monocytes",
+        "b_cells",
+        "cd34",
+        "cd56_nk",
+        "cd4_t_helper",
+        "regulatory_t",
+        "naive_t",
+        "memory_t",
+        "cytotoxic_t",
+        "naive_cytotoxic",
+    ],
+    "2.1.0": ["pbmc8k", "pbmc4k", "t_3k", "t_4k", "neuron_9k"],
+    "3.0.0": [
+        "pbmc_1k_protein_v3",
+        "pbmc_10k_protein_v3",
+        "malt_10k_protein_v3",
+        "pbmc_1k_v2",
+        "pbmc_1k_v3",
+        "pbmc_10k_v3",
+        "hgmm_1k_v2",
+        "hgmm_1k_v3",
+        "hgmm_5k_v3",
+        "hgmm_10k_v3",
+        "neuron_1k_v2",
+        "neuron_1k_v3",
+        "neuron_10k_v3",
+        "heart_1k_v2",
+        "heart_1k_v3",
+        "heart_10k_v3",
+    ],
+}
+
+dataset_to_group = dict(
+    [
+        (dataset_name, group)
+        for group, list_datasets in available_datasets.items()
+        for dataset_name in list_datasets
+    ]
+)
 group_to_url_skeleton = {
-    '1.1.0': "http://cf.10xgenomics.com/samples/cell-exp/{}/{}/{}_{}_gene_bc_matrices.tar.gz",
-    '2.1.0': "http://cf.10xgenomics.com/samples/cell-exp/{}/{}/{}_{}_gene_bc_matrices.tar.gz",
-    '3.0.0': "http://cf.10xgenomics.com/samples/cell-exp/{}/{}/{}_{}_feature_bc_matrix.tar.gz"
+    "1.1.0": "http://cf.10xgenomics.com/samples/cell-exp/{}/{}/{}_{}_gene_bc_matrices.tar.gz",
+    "2.1.0": "http://cf.10xgenomics.com/samples/cell-exp/{}/{}/{}_{}_gene_bc_matrices.tar.gz",
+    "3.0.0": "http://cf.10xgenomics.com/samples/cell-exp/{}/{}/{}_{}_feature_bc_matrix.tar.gz",
 }
-available_specification = ['filtered', 'raw']
+available_specification = ["filtered", "raw"]
 
 
-class Dataset10X(GeneExpressionDataset):
-    r""" Loads a file from `10x`_ website.
+class Dataset10X(DownloadableDataset):
```

Comment:
There's a minor issue here where the datasets with "protein" in the name contain columns that are proteins and not genes. We can parse the features because they indicate if a column is a "Gene" or "Antibody Capture". If a gene has the word "total" in it and is "Antibody Capture" we should store those as another cell measurement. Else if it has "Antibody Capture" and something other than "total" it should be stored as a miscellaneous cell measurement.

Reply:
We can probably also add a few lines to remove genes that are all 0.

Reply:
"We can probably also add a few lines to remove genes that are all 0." 
Let's not do that, it's not user friendly to get your dataset altered. The behavior with the cell removed is already not perfect

Reply:
I think we need to address this somehow though. A lot of the datasets from 10x have thousands of empty columns.

Reply:
these should not be selected when you apply gene filtering right ?

Reply:
It looks like if a column is all zeros, the standardscaler gives it a scale of 1. This could be problematic.

Reply:
oh that I did not know, then I think we should modify the gene_filter function but not the data loading API, right ?

Reply:
```python
scaler.fit(np.zeros((10, 10)))
Out[9]: StandardScaler(copy=True, with_mean=True, with_std=True)
scaler.var_
Out[10]: array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])
```

Reply:
what do you mean @adamgayoso ?

Reply:
I agree that we should try to isolate the protein data and store them as ``CellMeasurements`` . For the rest, the user can filter if he wants, apart from empty cells he's solely responsible of what's in the dataset when he trains!

Reply:
```
In [61]: scaler = StandardScaler(with_mean=False)                                                                                                                                                                                   

In [62]: scaler.fit(np.zeros((10, 10))) 
    ...:                                                                                                                                                                                                                            
Out[62]: StandardScaler(copy=True, with_mean=False, with_std=True)

In [63]: scaler.var_                                                                                                                                                                                                                
Out[63]: array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.])

In [64]: scaler.scale_                                                                                                                                                                                                              
Out[64]: array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.])
```

Reply:
We use `.var_` so should be fine. Very weird behavior.